app/controllers/serviceLearning/routes.py

```diff
@@ -87,9 +87,17 @@ def slcEditProposal(courseID):
 def slcCreateCourse():
     """will give a new course ID so that it can redirect to an edit page"""
     course = createCourse(g.current_user)
-
     return redirect(url_for('serviceLearning.slcEditProposal', courseID = course.id))
 
+@serviceLearning_bp.route('/serviceLearning/canceledProposal', methods=['POST'])
+def slcCancelProposal():
+    courseID = request.form.get('courseID')
+    course = Course.get_by_id(courseID)
+    if not course.courseName and not course.courseAbbreviation:
+        CourseQuestion.delete().where(CourseQuestion.course_id == courseID).execute()
+        course.delete_instance()
```

Comment:
This is redundant if you are using delete and execute above. Since you already have a course, you could just use `delete_instance` without the query above.

Reply:
The reason for the query is because CourseQuestion is dependent with the Course row of the same ID. We can definitely delete_instance for CourseQuestion though.